cmp/grid/filter/GridFilterModel.js

```diff
@@ -96,53 +96,43 @@ export class GridFilterModel extends HoistModel {
 
     /**
      * Set / replace the filters for a given field.
-     * @param {string} field - field to identify this filter
-     * @param {(Filter|Object|[])} filter - Filter(s), or config to create. If null, the filter will be removed
+     * @param {(Filter|Object|[])} filter - Filter(s), or config to create.
      */
     @action
-    setColumnFilters(field, filter) {
-        const currFilter = this.filter,
-            currFilters = currFilter?.isCompoundFilter ? currFilter.filters : [currFilter];
-
-        // Strip out any existing filters for this field
-        const newFilters = currFilters.filter(filter => {
-            if (!filter) return false;
-            if (filter.isFieldFilter && filter.field === field) return false;
-            if (filter.isCompoundFilter && find(filter.filters, it => it.field === field)) return false;
-            return true;
-        });
-
-        // Add in new filter
-        newFilters.push(...castArray(filter));
-        this.setFilter(newFilters);
+    setColumnFilters(filter) {
+        const ret = this.filter?.withFilter(filter) ?? filter;
```

Comment:
I guess the constant need for this null handling does argue for the utility function you originally wanted tom